### PR TITLE
KYLIN-4511 Support project admin user to calculate column cardinality…

### PR DESCRIPTION
**Pull request by guangxuCheng**
… by web

## Proposed changes

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce to Kylin?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have create an issue on [Kylin's jira](https://issues.apache.org/jira/browse/KYLIN), and have described the bug/feature there in detail
- [x] Commit messages in my PR start with the related jira ID, like "KYLIN-0000 Make Kylin project open-source"
- [x] Compiling and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] If this change need a document change, I will prepare another pr against the `document` branch
- [x] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at user@kylin or dev@kylin by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: webapp/app/js/controllers/sourceMeta.js [hunk 1]
```diff
@@ -131,6 +131,39 @@ KylinApp
...
+        text: "Are you sure to recalculate column cardinality?",
+        showCancelButton: true,
+        confirmButtonColor: '#DD6B55',
+        confirmButtonText: "Yes",
+        cancelButtonText: "No",
+        closeOnConfirm: true
+      }, function (isConfirm) {
+        if (isConfirm) {
+          if (!$scope.projectModel.selectedProject) {
+            SweetAlert.swal('', 'Please choose your project first!.', 'info');
```
Comment (shaofengshi) [inline webapp/app/js/controllers/sourceMeta.js:146]: The message should be consistent with other parts in below: "Please select a project." And, please avoid using "!" in the sentence as its mood is too strong.

File: webapp/app/js/controllers/sourceMeta.js [hunk 2]
```diff
@@ -131,6 +131,39 @@ KylinApp
...
+            return;
+          }
+          loadingRequest.show();
+          TableService.genCardinality({tableName: tableName, pro: $scope.projectModel.selectedProject}, {}, function () {
+            loadingRequest.hide();
+            MessageBox.successNotify('Cardinality job was calculated successfully. Please wait a moment...');
```
Comment (shaofengshi) [inline webapp/app/js/controllers/sourceMeta.js:152]: Some suggestions on the sentence: 'Cardinality job has been submitted successfully. Please wait a while to get the numbers.'

File: webapp/app/js/controllers/sourceMeta.js (hunk 1 above)
Comment (guangxuCheng) [reply webapp/app/js/controllers/sourceMeta.js:146]: OK, 'Please choose your project first!.' is also used in the other two places. I will deal with them together

File: webapp/app/js/controllers/sourceMeta.js (hunk 2 above)
Comment (guangxuCheng) [reply webapp/app/js/controllers/sourceMeta.js:152]: done

**Comment by guangxuCheng** [comment]
The latest code cannot be obtained here after using a force-push. If it is the same tomorrow, I will close the PR and create a new one. Thanks

**Comment by guangxuCheng** [comment]
@shaofengshi . Any other concerns? Thanks

**Comment by shaofengshi** [comment]
@guangxuCheng Looks good to me; Will merge it soon, thank you!

**Comment by RupengWang** [comment]
### Check list
- [x] Step 1. [**OPTIONAL**]Understand background/root cause/design basically(one hour EST) . Its issue type is 
    - [ ] bug fix
    - [ ] new feature
    - [x] enhancement
- [x] Step 2. Test cases are designed and documented (30 minutes EST).
- [x] Step 3. Prepare specific env, for example:
    - mock data (maybe some specific data type)
    - test env (maybe install a RDBMS instance).
- [x] Step 4. Verify and make sure test cases passed.
- [x] Step 5. Paste manual important steps and screenshots here (20 minutes EST). 
    - If you find difficulty in pick most important evidence, please attach diagnosis package. 
- [x] Step 6. Do more check in user perspective (20 minutes EST)
    - [x] Doc need be updated? And if it is updated? Ask help for release manager if so.
    - [x] If it is a breaking change so we should notify Kylin community? Ask help for release manager if so.
- [x] Step 7.  Summarize this test (20 minutes EST).

### Total estimate
- 4 hours for small issue from optimistic estimate. 

### Note
If you find some unexpected and **unrelated** error/mistake, please DO report it if it is truly a mistake, I think we may research and fix it in the future.

If you find background information/root cause analysis is not complete or ambiguous, please try to contact to author or do a quick research and record what you find. It is a good chance to learn something interesting.

If you find it is not easy to design testcase, please notify release manager.

**Comment by RupengWang** [comment]
## Design test cases

### Prepare test env
- Load one table from hive

### Case-1
1. 通过"Cardinality"按钮确定该功能正常可用并记录下每一行的Cardinality
2. Unload这个表
2. 通过System -> "Calculate Cardinality" 输入该表，然后验证两次的Cardinality是否相同

**Comment by RupengWang** [comment]
## Prepare data and env

- [x] Built on commit_SHA : 5b7b0ed1bdb49cb788e95207ca8437a05fa6b6d1
- [x] Hadoop env : HDP2.4
- [x] Just sample table is OK.

**Comment by RupengWang** [comment]
### Verify test case
#### Case-1
1. project级别的"Cardinality"
image
image
2. admin级别的System -> Calculate Cardinality
image

**Pull request merged**